libraries/chain/include/eosio/chain/peer_keys_db.hpp

```diff
@@ -13,34 +13,34 @@ namespace eosio::chain {
  */
 class peer_keys_db_t {
 public:
-   struct v0_data {                   // must match the one in eosio.system.hpp
-      std::optional<public_key_type> pubkey;
-   };
-
-   using peer_key_map_t = boost::unordered_flat_map<name, public_key_type, std::hash<name>>;
+   using peer_key_map_t = boost::unordered_flat_map<name, peer_info_t, std::hash<name>>;
+   using new_peers_t    = flat_set<name>;
 
    peer_keys_db_t();
 
    void set_active(bool b) { _active = b; }
 
    // must be called from main thread
-   size_t update_peer_keys(const controller& chain, uint32_t lib_number);
-
-   // safe to be called from any thread
-   std::optional<public_key_type> get_peer_key(name n) const;
+   // return the new peers either:
+   //  - added to the top-50
```

Comment:
All the comments about "top-50" should be updated to just say top according to what contract returns.